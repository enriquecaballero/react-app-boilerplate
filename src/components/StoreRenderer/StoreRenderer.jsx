import React from "react";

export default function StoreRenderer (props) {
  return <pre children={`redux: ${JSON.stringify (props.state, null, 2)}`} />;
}

StoreRenderer.defaultProps = {
  state: {}
};
