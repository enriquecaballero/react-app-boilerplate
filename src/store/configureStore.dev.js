import { createStore, compose, applyMiddleware } from "redux";
import rootReducer from "../reducers/";
import { persistStore, autoRehydrate } from "redux-persist";
import logger from "redux-logger";

const store = createStore (
  rootReducer,
  compose (applyMiddleware (logger), autoRehydrate ())
);

if (module.hot) {
  module.hot.accept ("../reducers/", () =>
    store.replaceReducer (require ("../reducers/"))
  );
}

export const persistor = persistStore (store);

export default store;
