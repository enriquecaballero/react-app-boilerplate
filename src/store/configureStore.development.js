import { createStore, compose, applyMiddleware } from "redux";
import rootReducer from "../reducers/";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import logger from "redux-logger";

const persistConfig = {
  key: "root",
  storage: storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, compose(applyMiddleware(logger)));

if (module.hot) {
  module.hot.accept("../reducers/", () =>
    store.replaceReducer(require("../reducers/"))
  );
}

export const persistor = persistStore(store);

export default store;
