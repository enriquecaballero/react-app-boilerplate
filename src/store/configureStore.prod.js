import { createStore, compose } from "redux";
import rootReducer from "../reducers/";
import { persistStore, autoRehydrate } from "redux-persist";

const store = createStore (rootReducer, compose (autoRehydrate ()));

export const persistor = persistStore (store);

export default store;
