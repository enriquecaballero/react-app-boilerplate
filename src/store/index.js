const getStoreConfiguration = () => {
  switch (process.env.NODE_ENV) {
    case "production":
    case "development":
      return require(`./configureStore/${process.env.NODE_ENV}`);
    default:
      return require("./configureStore/testing");
  }
};

module.exports = getStoreConfiguration();
