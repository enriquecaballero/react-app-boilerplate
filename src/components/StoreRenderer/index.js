import { connect } from "react-redux";
import StoreRenderer from "./StoreRenderer";

function mapStateToProps (state): Object {
  return {
    state
  };
}

export default connect (mapStateToProps) (StoreRenderer);
