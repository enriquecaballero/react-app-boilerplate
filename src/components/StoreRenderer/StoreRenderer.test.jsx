import React from "react";
import renderer from "react-test-renderer";
import store from "../../store";
import StoreRenderer from "./StoreRenderer";

describe("<StoreRenderer />", () => {
  it("works", () => {
    const component = renderer.create(
      <StoreRenderer state={store.getState()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
