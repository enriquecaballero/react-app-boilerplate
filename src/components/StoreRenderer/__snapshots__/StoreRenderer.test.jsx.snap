// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreRenderer /> works 1`] = `
<pre>
  redux: {
  "name": "World"
}
</pre>
`;
